scripts/setup-ssh-keys.sh

```
@@ -0,0 +1,767 @@
+    fi
+    
+    if [ ${#hosts[@]} -eq 0 ]; then
+        print_warning "No valid hosts found in SSH config"
+        return 1
+    fi
+    
+    # Apply host filter if specified
+    if [ -n "$HOST_FILTER" ]; then
+        local filtered_hosts=()
+        for host_entry in "${hosts[@]}"; do
+            local host_name=$(echo "$host_entry" | cut -d'|' -f1)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)